src/univariate/affine.jl

```diff
@@ -0,0 +1,332 @@
+        return d.σ * scale(d.ρ)
+    else
+        return d.σ
+    end
+end
+shape(d::AffineDistribution) = shape(d.ρ)
+ncategories(d::AffineDistribution) = ncategories(d.ρ)
+params(d::AffineDistribution) = (d.μ, d.σ, d.ρ)
+partype(::AffineDistribution{F,S,Tμ,Tσ,D}) where {F,S,Tμ,Tσ,D} = (Tμ, Tσ)
+
+Base.sign(d::UnivariateAffine) = Base.sign(d.σ)
+
+
+#### Statistics
+
+mean(d::AffineDistribution) = d.μ + d.σ * mean(d.ρ)
+median(d::AffineDistribution) = d.μ + d.σ * median(d.ρ)
+mode(d::AffineDistribution) = d.μ + d.σ * mode(d.ρ)
+modes(d::AffineDistribution) = d.μ .+ d.σ .* modes(d.ρ)
+
+var(d::UnivariateAffine) = d.σ^2 * var(d.ρ)
+std(d::UnivariateAffine) = abs(d.σ) * std(d.ρ)
+skewness(d::UnivariateAffine) = Base.sign(d) * skewness(d.ρ)
+kurtosis(d::UnivariateAffine) = kurtosis(d.ρ)
+
+cov(d::AffineDistribution) = d.σ * cov(d.ρ) * d.σ'
+
+isplatykurtic(d::AffineDistribution) = isplatykurtic(d.ρ)
+isleptokurtic(d::AffineDistribution) = isleptokurtic(d.ρ)
+ismesokurtic(d::AffineDistribution) = ismesokurtic(d.ρ)
+
+log_abs_det(x::AbstractMatrix) = first(logabsdet(x))
+log_abs_det(x::Real) = log(abs(x))
+
+entropy(d::ContinuousAffine) = entropy(d.ρ) + log_abs_det(d.σ)
+entropy(d::DiscreteAffine) = entropy(d.ρ)
+
+mgf(d::AffineDistribution, t::Real) = exp(d.μ * t) * mgf(d.ρ, d.σ * t)
+cf(d::AffineDistribution, t::Real) = cf(d.ρ, t * d.σ) * exp(1im * t * d.μ)
+
+
+#### Evaluation & Sampling
+
+# Continuous
+function pdf(d::ContinuousAffine{Univariate}, x::Real) 
+    return pdf(d.ρ, d.σ \ (x - d.μ)) / abs(d.σ)
+end
+function pdf(d::ContinuousAffine{ArrayLikeVariate{N}}, x::AbstractArray{N}) where N
+    return pdf(d.ρ, d.σ \ (x - d.μ)) / abs(det(d.σ))
+end
+
+function logpdf(d::ContinuousAffine{Univariate}, x::Real) 
+    return logpdf(d.ρ, d.σ \ (x - d.μ)) - log_abs_det(d.σ)
+end
+function logpdf(d::ContinuousAffine{ArrayLikeVariate{N}}, x::AbstractArray{N}) where N
+    return logpdf(d.ρ, d.σ \ (x - d.μ)) - log_abs_det(d.σ)
+end
+
+# Discrete
+
+function pdf(d::DiscreteAffine{Univariate}, x::Real)
+    return pdf(d.ρ, d.σ \ (x - d.μ))
+end
+function pdf(d::DiscreteAffine{ArrayLikeVariate{N}}, x::AbstractArray{N}) where N
+    return pdf(d.ρ, d.σ \ (x - d.μ))
+end
+function logpdf(d::DiscreteAffine{Univariate}, x::Real)
+    return logpdf(d.ρ, d.σ \ (x - d.μ))
+end
+function logpdf(d::DiscreteAffine{ArrayLikeVariate{N}}, x::AbstractArray{N}) where N
+    return logpdf(d.ρ, d.σ \ (x - d.μ))
+end
+
+# CDF methods
+
+function cdf(d::UnivariateAffine, x::Real)
+    x = (x - d.μ) / d.σ
+    return d.σ < 0 ? ccdf(d.ρ, x) : cdf(d.ρ, x)
+end
+
+function logcdf(d::UnivariateAffine, x::Real)
+    x = (x - d.μ) / d.σ
+    return d.σ < 0 ? logccdf(d.ρ, x) : logcdf(d.ρ, x)
+end 
+
+function quantile(d::UnivariateAffine, q::Real)
+    q = d.σ < 0 ? (1 - q) : q
+    return d.μ + d.σ * quantile(d.ρ, q)
+end
+
+rand(rng::AbstractRNG, d::AffineDistribution) = d.μ + d.σ * rand(rng, d.ρ)
+
+function gradlogpdf(d::ContinuousAffine, x::Real)
```

Comment:
We should restrict it to univariate distributions.

Reply:
Why?

Reply:
What's wrong with it?